cpp/include/cudf/io/text/data_chunk_source_factories.hpp

```diff
@@ -136,6 +136,49 @@ class istream_data_chunk_reader : public data_chunk_reader {
   std::vector<host_ticket> _tickets;
 };
 
+/**
+ * @brief a reader which produces views of device memory which contain a copy of the data from a
+ * string
+ *
+ */
+class host_span_data_chunk_reader : public data_chunk_reader {
+ public:
+  host_span_data_chunk_reader(cudf::host_span<const char> data) : _data(data) {}
+
+  void skip_bytes(std::size_t read_size) override
+  {
+    _position += std::min(read_size, _data.size() - _position);
+  }
+
+  std::unique_ptr<device_data_chunk> get_next_chunk(std::size_t read_size,
+                                                    rmm::cuda_stream_view stream) override
+  {
+    CUDF_FUNC_RANGE();
```

Comment:
Do these need to be defined in the header?
I think we could slim down the header includes, etc if the definitions were in a .cpp file.

Reply:
Definitely don't need to be in the header :)
https://github.com/rapidsai/cudf/pull/10325/files#diff-e9af0335302cf69b508c8aafb5f334f8f3b4817006e01dac528ce079309e6f9b +
https://github.com/rapidsai/cudf/pull/10325/files#diff-920a2edfa1c5ed94bdd1c91341c6c499c30f1e70283fa306e5272033d5201e00